packages/vault/src/index.ts

```diff
@@ -58,14 +58,62 @@ export class VaultEnvVars {
     }
 
     this.vaultPath = options
+    this.vaultTokenFile = path.join(os.homedir(), '.vault-token')
   }
 
   async get(): Promise<Secrets> {
     const token = await this.getAuthToken()
     return this.fetchSecrets(token)
   }
 
+  private async fetchTest(token: string): Promise<boolean> {
+    try {
+      await fetch<ReturnFetch>(`${VAULT_ADDR}/secret?help=1`, {
+        headers: { 'X-Vault-Token': token }
+      })
+      return true
+    } catch {
+      return false
+    }
+  }
+
+  private async handleLocalToken(): Promise<string> {
+    try {
+      console.log('checking for current token')
+      const vaultToken = await fs.readFile(this.vaultTokenFile, {
+        encoding: 'utf8'
+      })
+      console.log('testing current token')
+      const validToken = await this.fetchTest(vaultToken)
+      if (validToken) {
+        console.log('success!')
+        return vaultToken
+      } else {
+        throw new ToolKitError('current token invalid, requesting new one...')
```

Comment:
**suggestion:** if line 92 or line 95 executes, I'm not sure if their error would be output as it might be swallowed by the catch block in line 135. For instance, if line 92 executes and therefore `new ToolKitError('current token invalid, requesting new one...')` is thrown, and then in line 136 `VAULT_AUTH_GITHUB_TOKEN` is undefined, we will end up with the error 'VAULT_AUTH_GITHUB_TOKEN variable is not set' with the 'current token invalid, requesting new one...'. being swallowed. To guarantee the user can see the error from line 92 / 95, I think we could log out the error here as well

```suggestion
        const message = 'current token invalid, requesting new one...'
        console.error(message)
        throw new ToolKitError(message)
```